MobileWallet/Libraries/TariLib/Core/FFI/Keys/TariAddress.swift

```diff
@@ -104,17 +159,71 @@ final class TariAddress {
 }
 
 extension TariAddress: Equatable {
-
     static func == (lhs: TariAddress, rhs: TariAddress) -> Bool {
         guard let leftHex = try? lhs.byteVector.hex, let rightHex = try? rhs.byteVector.hex else { return false }
         return leftHex == rightHex
     }
+}
+
+extension TariAddress {
 
+    @available(*, deprecated, message: "This getter is obsolete and it will be removed in the future.")
     var publicKey: String {
         get throws { try String(byteVector.hex.dropLast(2)) }
     }
 
     var isUnknownUser: Bool {
         get throws { try publicKey.filter { $0 == "0" }.count == 64 }
     }
+
+    static func makeTariAddress(input: String) throws -> TariAddress {
+        do { return try TariAddress(emojiID: input) } catch {}
+        return try TariAddress(base58: input)
+    }
+
```

Comment:
Do you use this method for address validating?

Reply:
It's universal method to create TariAddress using "any" string. If input value is valid then this factory will make an object, if don't it will thrown an error.